interfaces/Portalicious/src/app/pages/user-settings/user-settings.component.ts

```diff
@@ -1,12 +1,99 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, signal } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  ValidatorFn,
+  Validators,
+} from '@angular/forms';
+import { ButtonModule } from 'primeng/button';
+import { PasswordModule } from 'primeng/password';
 import { PageLayoutComponent } from '~/components/page-layout/page-layout.component';
 
 @Component({
   selector: 'app-user-settings',
   standalone: true,
-  imports: [PageLayoutComponent],
+  imports: [
+    PageLayoutComponent,
+    PasswordModule,
+    ButtonModule,
+    ReactiveFormsModule,
+  ],
   templateUrl: './user-settings.component.html',
   styles: ``,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserSettingsComponent {}
+export class UserSettingsComponent {
+  changePasswordForm = new FormGroup(
+    {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      currentPassword: new FormControl('', [Validators.required]),
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      newPassword: new FormControl('', [Validators.required]),
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      confirmPassword: new FormControl('', [Validators.required]),
+    },
+    {
+      validators: this.matchValidator('password', 'confirmPassword'),
+    },
+  );
+
+  matchValidator(
+    controlName: string,
+    matchingControlName: string,
+  ): ValidatorFn {
+    return (abstractControl: AbstractControl) => {
+      const control = abstractControl.get(controlName);
+      const matchingControl = abstractControl.get(matchingControlName);
+
+      if (
+        matchingControl?.errors &&
+        !matchingControl.errors.confirmedValidator
+      ) {
+        return null;
+      }
+
+      if (control?.value !== matchingControl?.value) {
+        const error = { confirmedValidator: 'Passwords do not match.' };
```

Comment:
```suggestion
        const error = {
          confirmedValidator: $lokalise`Passwords do not match.`
        };
```

Or could we let this string 'come from' the template-file?
So with an additional attribute on the password-field or something?